decidim-core/app/controllers/concerns/decidim/form_factory.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require "active_support/concern"
+
+module Decidim
+  # Common logic to build form objects
+  module FormFactory
+    extend ActiveSupport::Concern
+
+    included do
+      def form(klass)
+        Class.new do
+          def initialize(klass, context)
+            @klass = klass
+            @context = context
+          end
+
+          def instance
+            @klass.new
+          end
+
+          def from_model(params)
+            @klass.from_model(params)
+          end
+
+          def from_params(params, context = {})
+            @klass.from_params(params, context_hash.merge(context))
+          end
+
+          def context_hash
+            {
+              current_organization: @context.try(:current_organization),
+              current_user: @context.try(:current_user)
+            }
```

Comment:
Maybe we should delete nil keys? Also, why `try`? I mean, when is valid to pass a nil context to the factory? Maybe we should raise if klass or context are nil.